com.unity.perception/Runtime/GroundTruth/RenderTextureReader.cs

```diff
@@ -34,8 +34,7 @@ public RenderTextureReader(RenderTexture source, Camera cameraRenderingToSource,
             this.m_CameraRenderingToSource = cameraRenderingToSource;
             m_NextFrameToCapture = Time.frameCount;
 
-            if (!GraphicsUtilities.SupportsAsyncReadback())
-                m_CpuTexture = new Texture2D(m_Source.width, m_Source.height, m_Source.graphicsFormat, TextureCreationFlags.None);
+            m_CpuTexture = new Texture2D(m_Source.width, m_Source.height, m_Source.graphicsFormat, TextureCreationFlags.None);
```

Comment:
It's hard to know when someone may choose to go into visualization mode, and right now, we are creating the cpu texture as a part of initialization. So either we need to create the cpu texture and take the chance that it may never be used, or check if it has been allocated before we use it each time.

Reply:
At the moment this type still only uses `GraphicsUtilities.SupportsAsyncReadback()` to determine whether it should use m_CpuTexture. Were you intending some other changes here to go along with the visualization toggle?

Reply:
Changed it to check if `m_CpuTexture == null` and allocating it before accessing it.